src/components/Editor/ItemEditor.vue

```diff
@@ -734,6 +779,18 @@ export default {
         this.selectedItemDetail.max_char_limit = value;
       },
     },
+    isThisSurveyQuestion: {
+      get() {
+        // extract whether question is for survey
+        if (this.selectedItemDetail == null) return false;
+        return this.selectedItemDetail.survey;
+      },
+      set(value) {
+        // set whether question is for survey
+        this.isSurveyQuestion = true;
+        this.selectedItemDetail.survey = value;
+      },
+    },
```

Comment:
For this, you don't need to add a getter and setter. Almost always (99% of times) try to keep computed properties READ-ONLY.
Maybe there are a lot of getter/setter in computed properties in this file and I think that is a bad design that we made when we were writing this code.

You want to **compute** if the current selected item is a survey question or not. And after doing that computation, you want to store the result in a variable called `isThisSurveyQuestion`. That should be the only purpose of this variable. A read only variable. 

But then what would you have to do to **re-compute** the value of `isThisSurveyQuestion` if someone clicks on the checkbox?
**you would just change the `selectedItemDetails.survey` to true lets say. And because `isThisSurveyQuestion` is reactive, it would pick up the changes and recompute it's value**


So two changes
- Remove getter, setter. Only the getter functionality will remain
- In the HTML where you're doing `v-model="isThisSurveyQuestion"`, you should directly do `v-model="selectedItemDetail.survey"`.